src/components/MonthCalendar/MonthCalendar.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react'
+import {
+  GridList,
+  GridListItem,
+} from 'react-aria-components'
+import { CalendarDate, getLocalTimeZone, ZonedDateTime, fromDate } from '@internationalized/date'
+import { ChevronLeft, ChevronRight } from 'lucide-react'
+import { HStack, VStack } from '../ui/Stack/Stack'
+import { Button } from '../ui/Button/Button'
+import { Heading } from '../ui/Typography/Heading'
+import { MONTHS } from './utils'
+import './monthCalendar.scss'
+
+export function MonthCalendar({
+  date,
+  onChange,
+  minDate,
+  maxDate,
+}: {
+  date: ZonedDateTime | null
+  onChange: (val: ZonedDateTime) => void
+  minDate?: ZonedDateTime | null
+  maxDate?: ZonedDateTime | null
+}) {
+  const minYear = minDate?.year ?? null
+  const maxYear = maxDate?.year ?? null
+  const [year, setYear] = useState(() => date?.year ?? new Date().getFullYear())
+
+  function clampYear(y: number) {
+    if (minYear != null && y < minYear) return minYear
+    if (maxYear != null && y > maxYear) return maxYear
+    return y
+  }
+
+  function isMonthDisabled(month: number): boolean {
+    const date = new CalendarDate(year, month, 1)
+
+    if (minDate) {
+      const minCalendarDate = new CalendarDate(minDate.year, minDate.month, 1)
+      if (date.compare(minCalendarDate) < 0) return true
+    }
+
+    if (maxDate) {
+      const maxCalendarDate = new CalendarDate(maxDate.year, maxDate.month, 1)
+      if (date.compare(maxCalendarDate) > 0) return true
+    }
```

Comment:
I _think_ this works right but I'm squinting here- does this allow september to be selectable if the max date is september 15? I _think_ so but the `> 0` instead of `>= 0` makes me nervous